### ISSUE TITLE
First Powershell script in MAXfocus

### ISSUE THREAD

**Issue by techinaflash** [klemmestad/PowerShell#1]
Looks like some very hard work went into these scripts. I'd like to make use of them.  Is there something specific that needs to be done to each computer on MAXfocus to utilize a Powershell script.  I'm new to Powershell, and have had some issues getting my short/simple scripts to run.  I'm getting errors about setting the execution policy.  I'm not sure if that needs to be done if you are running the scripts through the dashboard as it's supposed to run as a local system account.  If it does, do you have a way to automate it?

**Comment by klemmestad** [comment]
No, you do not have to do anything specific. The scripts will be run regardless of executionpolicy, as the agent modifies the executionpolicy on the fly. You only need to modify the scripts to your liking and upload them as user scripts to the dashboard. Apply them as script checks or automated tasks as it suits your workflow. I have tried hard to make them well behaved enough to execute as checks, but it makes no difference to the script which deployment method you choose. With workstations I use Site Maintenance Tasks, with servers I use checks. And I use checks because the dashboard allows me to add and modify checks on multiple servers in one go. Automated tasks must be added or edited on one device at at time. The downside of checks is that 150 seconds is maximum allowed runtime. It the script isn't done by then, it will be killed. On older servers that makes autoconfiguration a challenge, but  on most modern servers it is enough.

**Comment by jonathan-roy-8** [comment]
I've try to execute your script following the following [video ](https://www.youtube.com/watch?v=M-396UzP_BE) and when i try to execute your script in powershell ISE, it telling me 

Cannot call the method on a expression that has the value null
`Vous ne pouvez pas appeler de méthode sur une expression ayant la valeur Null.
Au niveau de C:\snmppauto.ps1 : 301 Caractère : 55
- $gfimaxpath = Split-Path $gfimaxagent.PathName.Replace <<<< ('"',"") -Parent
  - CategoryInfo          : InvalidOperation: (Replace:String) [], ParentContainsErrorRecordException
  - FullyQualifiedErrorId : InvokeMethodOnNull`

**Comment by klemmestad** [comment]
That only occurs if the $gfimaxagent variable is  $null. This is because it could not find any service named "advanced monitoring agent".

Any chance the monitoring agent is named differently when localized in French? If so, you must change the service name in the line above the one that fails.

Hope that helps!

Med vennlig hilsen
Hugo L. Klemmestad
(@mobile)

Den 19. mar. 2016 kl. 00.00 skrev jonathan-roy-8 <notifications@github.com<mailto:notifications@github.com>>:

I've try to execute your script following the following video https://www.youtube.com/watch?v=M-396UzP_BE and when i try to execute your script in powershell ISE, it telling me

Cannot call the method on a expression that has the value null
`Vous ne pouvez pas appeler de m?thode sur une expression ayant la valeur Null.
Au niveau de C:\snmppauto.ps1 : 301 Caract?re : 55
-   $gfimaxpath = Split-Path $gfimaxagent.PathName.Replace <<<< ('"',"") -Parent
  -   CategoryInfo : InvalidOperation: (Replace:String) [], ParentContainsErrorRecordException
  -   FullyQualifiedErrorId : InvokeMethodOnNull`

##
